### PR TITLE
feat: Logger using zerolog

### LINKED ISSUE #46: Support zerolog as logging
**Is your feature request related to a problem? Please describe.**

- zerolog has the best performance in logger libraries.
- zerolog is wide use.
- hertz/logger already support zerolog: https://www.cloudwego.io/docs/hertz/tutorials/observability/log/

**Describe the solution you'd like**

Support logging/zerolog package which well integrate to otel.

**Describe alternatives you've considered**

NO

### PR DESCRIPTION
**Pull request by lovung**
#### What type of PR is this?

feat: Logger using zerolog

#### What this PR does / why we need it (English/Chinese):

Close #46 

#### Which issue(s) this PR fixes:

Missing zerolog when integrate with OpenTelemetry

### CODE REVIEW

File: logging/zerolog/logger.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
+// Copyright 2022 CloudWeGo Authors.
```
Comment (li-jin-gou) [inline logging/zerolog/logger.go:1]: 2022 -> 2024

File: logging/zerolog/logger_test.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
+// Copyright 2022 CloudWeGo Authors.
```
Comment (li-jin-gou) [inline logging/zerolog/logger_test.go:1]: +1

File: logging/zerolog/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+)
+
+func OtelZerologMiddleware() app.HandlerFunc {
+	return func(ctx context.Context, reqCtx *app.RequestContext) {
+		ctx = log.Logger.WithContext(ctx)
+
+		span := trace.SpanFromContext(ctx)
+
+		ctx = context.WithValue(ctx, zerolog.ExtraKey(zerolog.TraceIDKey), span.SpanContext().TraceID())
+		ctx = context.WithValue(ctx, zerolog.ExtraKey(zerolog.SpanIDKey), span.SpanContext().SpanID())
```
Comment (rogerogers) [inline logging/zerolog/README.md:22]: Can you provide a more detailed description of this scene? In one request, there can be multiple spanids.

File: logging/zerolog/logger.go [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+		l.Logger.CtxWarnf(ctx, format, kvs...)
+	}
+
+	if !span.IsRecording() {
+		l.Logger.Logf(level, format, kvs...)
+		return
+	}
+
+	// set span status
+	if zlevel >= l.config.traceConfig.errorSpanLevel {
```
Comment (rogerogers) [inline logging/zerolog/logger.go:98]: The log level of zerolog needs more evaluation.

File: logging/zerolog/logger_test.go [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+	logger := NewLogger(
+		WithTraceErrorSpanLevel(zerolog.WarnLevel),
+		WithRecordStackTraceInSpan(true),
+	)
+
+	// output to buffer
+	logger.SetOutput(buf)
+
+	logger.Debug("this is a debug log")
+	assert.NotContains(t, buf.String(), "this is a debug log")
```
Comment (rogerogers) [inline logging/zerolog/logger_test.go:121]: The default log level of zerolog is TraceLevel.

File: logging/zerolog/logger.go (hunk 1 above)
Comment (lovung) [reply logging/zerolog/logger.go:1]: I fix modify

File: logging/zerolog/README.md (hunk 3 above)
Comment (lovung) [reply logging/zerolog/README.md]: Updated it. It is not necessary to have the middleware anymore.

File: logging/zerolog/logger.go [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+		Logger: *newLogger,
+		config: config,
+	}
+}
+
+func (l *Logger) CtxLogf(level hlog.Level, ctx context.Context, format string, kvs ...any) {
+	var zlevel zerolog.Level
+	span := trace.SpanFromContext(ctx)
+
+	ctx = context.WithValue(ctx, ExtraKey(TraceIDKey), span.SpanContext().TraceID())
```
[the remark this answers is not in the document]
Comment (lovung) [reply logging/zerolog/logger.go]: Updated it to check. But for traceFlags, don't have a way to check it.

Anyways, we use the same code in other log package in this repo.

File: logging/zerolog/logger.go (hunk 4 above)
Comment (lovung) [reply logging/zerolog/logger.go:89]: I think this check is OK. By default, we use ErrorLevel as errorSpanLevel.

File: logging/zerolog/logger_test.go (hunk 5 above)
Comment (lovung) [reply logging/zerolog/logger_test.go:126]: Fixed unit test.

File: logging/zerolog/logger.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (rogerogers) [reply logging/zerolog/logger.go]: OK

**Review by rogerogers — APPROVED** [review]

**Pull request merged**